### DISCUSSION TITLE
[FR] Use middle mouse button to open note in new tab

### DISCUSSION THREAD

**Discussion by Moyf** [chrisgurney/obsidian-note-toolbar#101] [Ideas]
As the update in [1.8.2](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.8.2) supports the modification keys (such as `ctrl`) to open note in new tab, could you also consider adding support for "Middle mouse button to open in new tab" (Similar to `Ctrl+Click`) ?
This feature would allow users to easily open notes in new tab using just the mouse, no need to use another hand to press the keyboard  : D

(And as far as I am aware, the middle click hasn't been assigned with other function for now? )

**Comment by chrisgurney** [comment]
@Moyf thanks for the idea!

Do markdown links in Obsidian notes open in a new tab with the middle mouse button, by default?

_Edit:_ Yes, looks like it's supposed to: https://forum.obsidian.md/t/middle-click-to-open-file-in-a-new-tab/59930/2

Since this is default behavior, I'll look into it!

**Reply by Moyf** [reply]
Glad to hear that !  😆

**Reply by chrisgurney** [reply]
@Moyf Got it working locally, though once it's released I'll need your help testing it to be sure. Look for it hopefully in the next day or two!

**Reply by Moyf** [reply]
> @Moyf Got it working locally, though once it's released I'll need your help testing it to be sure. Look for it hopefully in the next day or two!

Yay no problem!  ; )

**Reply by chrisgurney** [reply]
@Moyf can you please test [v1.9.4](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.9.4) and let me know if middle clicking now opens file items in tabs for you?

**Reply by Moyf** [reply]
> @Moyf can you please test [v1.9.4](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.9.4) and let me know if middle clicking now opens file items in tabs for you?

Yeah just updated to 1.9.5 and it works perfectly, thanks !!
(And I noticed the 🌞 in your release log hahahaha, praise the sun !

**Comment by Moyf** [comment]
Oh by the way, I noticed that NTB also supports URL link like `obsidian://note-toolbar?commandid=xxx`, it's really great!

And I saw in the document :
> You will have to figure out command IDs on your own in order for this to work (by inspecting an existing toolbar, or by [using Advanced URI](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/How-to-get-a-URI-with-Advanced-URI) for example). I know this is not ideal, so I will very likely add a Copy URI for command helper/command in the future.

It reminds me that I used to wrote a Templater script to select command and get the command-id:

```
<% (await tp.system.suggester((item) => item.name, app.commands.listCommands(f => f.name))).id %>
```

How it works:
![98f7f94a-e5bd-4587-9c78-4274e87cb0f5](https://github.com/user-attachments/assets/b26575a3-e7ae-458c-adcb-adbb565dfb33)

It's really simple but I hope it could be somewhat helpful

**Reply by chrisgurney** [reply]
Hey that's great! Thanks for sharing!

**Reply by chrisgurney** [reply]
@Moyf I've added your script to the documentation: https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Note-Toolbar-Callouts

**Reply by Moyf** [reply]
> @Moyf I've added your script to the documentation: https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Note-Toolbar-Callouts

My pleasure! 😏
